meta-resin-common/recipes-core/chrony/files/chrony.conf

```diff
@@ -1,8 +1,11 @@
-initstepslew 30 0.resinio.pool.ntp.org
 server 0.resinio.pool.ntp.org iburst
 server 1.resinio.pool.ntp.org iburst
 server 2.resinio.pool.ntp.org iburst
 server 3.resinio.pool.ntp.org iburst
 driftfile /var/lib/chrony/drift
-makestep 1 3
+maxupdateskew 100
+makestep 1 -1
+logchange 1
+maxdistance 16
+hwtimestamp *
```

Comment:
This is actually potentially really dangerous for us.

Our devices run at client sites and stream timeseries data back to our web apis.

`makestep 1 -1` means that chrony will at any time step the clock any amount if it sees that it's drifted from ntp time.

...which for us means gaps in the timeseries if the clock jumps forward (which is bad), or corrupted timeseries data if the clock jumps backward (which is catastrophic).

I would strongly prefer that makestep be a configurable option.

Reply:
Regarding backward jumps, chronyd won't jump back.
From https://chrony.tuxfamily.org/faq.html#_is_code_chronyd_code_allowed_to_step_the_system_clock
`applications started after chronyd will not observe backward steps`

Forward jumps can happen if the time difference is larger than 1 second. This is actually similar to the previous timesyncd functionality. In timesyncd, the clock was slewed only if time was different by 0.4seconds.
https://github.com/systemd/systemd/blob/master/src/timesync/timesyncd-manager.c#L46
Otherwise it was stepped. In this case, the clock will be skewed if the difference is 1 seconds. Otherwise it will step.

I agree with you that there might be some benefit in having configurable options there and will tag it as a feature request.

Reply:
👍🏻